License2Deploy/rolling_deploy.py

```diff
@@ -0,0 +1,215 @@
+      elif desired_state == 'decrease':
+        new_count = self.decrease_autoscale_instance_count(cur_count)
+      logging.info("Current desired count was changed from {0} to {1}".format(cur_count, new_count))
+      return new_count
+    except UnboundLocalError as u:
+      logging.error("Please make sure the desired_state is set to either increase or decrease: {0}".format(u))
+      exit(self.exit_error_code)
+ 
+  def double_autoscale_instance_count(self, count):
+    ''' Multiply current count by 2 '''
+    return count * 2
+
+  def decrease_autoscale_instance_count(self, count):
+    ''' Divide current count in half '''
+    return count / 2
+
+  def set_autoscale_instance_desired_count(self, new_count, group_name):
+    ''' Increase desired count by double '''
+    try:
+      logging.info("Set autoscale capacity for {0} to {1}".format(group_name, new_count))
+      self.conn_auto.set_desired_capacity(group_name, new_count)
+      return True
+    except Exception as e:
+      logging.error("Unable to update desired count, please investigate error: {0}".format(e))
+      exit(self.exit_error_code)
+      
+  def get_all_instance_ids(self, group_name):
+    ''' Gather Instance id's of all instances in the autoscale group '''
+    instances = [ i for i in self.get_group_info(group_name)[0].instances ]
+    id_list = []
+    for instance_id in instances:
+      id_list.append(instance_id.instance_id)
+
+    logging.info("List of all Instance ID's in {0}: {1}".format(group_name, id_list))
+    return id_list
+
+  def get_instance_ids_by_requested_build_tag(self, id_list, build):
+    ''' Gather Instance id's of all instances in the autoscale group '''
+    reservations = self.conn_ec2.get_all_reservations()
+    new_instances = []
+    for instance_id in id_list:
+      rslt = [inst for r in reservations for inst in r.instances if 'BUILD' in inst.tags and inst.id == instance_id]
+      for new_id in rslt:
+        if new_id.tags['BUILD'] == str(build):
+          new_instances.append(instance_id)
+    
+    if new_instances:
+      logging.info("New Instance List: {0}".format(new_instances))
+      return new_instances
+    else:
+      logging.error("New Instance List is empty, something went wrong")
+      exit(self.exit_error_code)
+    
+  def wait_for_new_instances(self, instance_ids, retry=9, wait_time=30):
+    ''' Monitor new instances that come up and wait until they are ready '''
+    for instance in instance_ids:
+      count = 0
+      health = []
+      while (count <= retry) and (len(health) < 2):
+        instanceStatus = self.conn_ec2.get_all_instance_status(instance)
+        for state in instanceStatus:
+          health = [x for x in [str(state.system_status.status), str(state.instance_status.status)] if x == "ok"]
+          if (len(health) < 2):
+            logging.warning("{0} is not in a fully working state yet".format(instance))
+            sleep(wait_time)
+            count = (count + 1)
```

Comment:
Minor aesthetic suggestions
Can reduce this to: 
``` python
count += 1
```
To keep with aesthetics used throughout this class, parenthesis are not necessary.
(From line right below) 
``` python 
if (count > retry):
```
can be:
``` python
if count > retry:
```